src/coreclr/jit/fginline.cpp

```diff
@@ -1654,14 +1654,12 @@ Statement* Compiler::fgInlinePrependStatements(InlineInfo* inlineInfo)
 
                     if (varTypeIsStruct(argType))
                     {
-                        structHnd = gtGetStructHandleIfPresent(argNode);
-                        noway_assert((structHnd != NO_CLASS_HANDLE) || (argType != TYP_STRUCT));
+                        structHnd = lclVarInfo[argNum].lclVerTypeInfo.GetClassHandleForValueClass();
+                        assert(structHnd != NO_CLASS_HANDLE);
```

Comment:
This is the reason for the few diffs: previously, the struct handle could be "refined" to a type more specific than `f(__Canon)`, which affects VN-based copy propagation.

The diffs could be fixed (i. e. made to not exist), but only with quirks, which I did not find worth it due to the already-not-small size of this change.